packages/flame/lib/src/game/mixins/game.dart

```diff
@@ -110,16 +110,29 @@ mixin Game on Loadable implements Projector {
       ''');
     }
     _gameRenderBox = gameRenderBox;
+    onAttach();
   }
 
+  /// Called when the game has been attached this can be overriden
+  /// to add logic that requires that the game to be already attached
+  /// to the widget tree
+  void onAttach() {}
```

Comment:
we could just add mustCallSuper to `atach` and then you can override attach, call super & do what you want on top. not sure which is better though, just a thought. def would need to update the lifecycle diagrams though

Reply:
That was my first attempt, but the parameters of the attach makes it impossible for external packages to override because it receives an inter an class from flame

Reply:
ah, good point. I do lean against having private classes though, if someone wants to use GameRenderBox for something, they should be allowed to. but regardless if we expose it I am ok with having onAttach as an easier/only alternative